dvc/repo/get.py

```diff
@@ -43,7 +44,10 @@ def get(url, path, out=None, rev=None):
     dpath = os.path.dirname(os.path.abspath(out))
     tmp_dir = os.path.join(dpath, "." + str(shortuuid.uuid()))
     try:
-        with external_repo(cache_dir=tmp_dir, url=url, rev=rev) as repo:
+        cached_clone(url, rev=rev, clone_path=tmp_dir)
```

Comment:
Sorry, I might've not worded it clearly enough, but I meant that we should keep it as it was before this change, meaning that we shouldn't pass tmp_dir to cached_clone and let it clone wherever it needs to. And we need to setup external cache dir pointing to tmp_dir, the same way we do that in external_repo. Also, take a look at `external_repo`, it is doing things like setting a default remote, which you've lost here.

Reply:
You might be able to flip the logic over and do something like:
```
try:
    with external_repo():
        ...
        if out.use_cache:
            _get_cached()
           return
except NotDvcRepoError:
     pass
path = cached_clone(...)
# copy from git
```

Reply:
Done! I didn't fold it like that because I would have to do the cached_clone conditionally, and instead duplicated the file copy call.